### DISCUSSION TITLE
How to close tabs

### DISCUSSION THREAD

**Discussion by schrute99** [LazyVim/LazyVim#531] [Q&A]
Hey all,

I'm trying to close tabs using the keymap `<leader><tab>d` but get the error message `E784: Cannot close last tab page`.
How can I close tabs using keymaps?

Thanks!

**Comment by dpetka2001** [comment]
I believe this is normal behavior. Since it's the last tab you should use `:q` to quit Lazyvim all together. Not sure if you can change this behavior. See also `:h tabc`

**Reply by schrute99** [reply]
Thanks for the answer. This also happens when I have multiple tabs open and try to close one of the tabs.

**Reply by dpetka2001** [reply]
The error you wrote in your first message is for when the tab you have is the last one. It doesn't show it when you have more than one tab. I just tested it and it works just fine.

**Reply by Darius3001** [reply]
For me it also doesnt work. Neither `<leader><tab>d` nor `:tabclose`.
Actually non of the tab operations work (like tabnext/ previous/ last etc.)
Just to be sure that we all talk about the same thing and im not in the wrong issue, im talking about these tabs:

![Screenshot from 2023-11-16 13-03-57](https://github.com/LazyVim/LazyVim/assets/20456547/107bf53d-5ead-43d6-9035-637635eca717)

**Reply by dpetka2001** [reply]
These are buffers, not tabs.

**Comment by octoxan** [comment]
Have the same problem, have like 4 tabs open on the left, and 2 tabs open on the right. I'm trying to close the duplicates so there 2 on the right are not also on the left, and each one tells it can't close the last tab

**Reply by dpetka2001** [reply]
You should read up on vim buffers/tabs terminology. The ones on the left are buffers and the ones on the right are tabs. Buffers are independent of tabs. When you close a tab, buffer does not close. You should also close the buffer for it not to appear. From my understanding, you should consider tabs as kind of different layouts of the buffers. Tabs do not provide a different workspace and should not be considered as such.

**Comment by boa-yang** [comment]
> For me it also doesnt work. Neither `<leader><tab>d` nor `:tabclose`. Actually non of the tab operations work (like tabnext/ previous/ last etc.) Just to be sure that we all talk about the same thing and im not in the wrong issue, im talking about these tabs:
> 
> ![Screenshot from 2023-11-16 13-03-57](https://user-images.githubusercontent.com/20456547/283442556-107bf53d-5ead-43d6-9035-637635eca717.png)

1. The tabs are on the top of the window's right side, there are some numbers like 1 ,2 , 3 ...(if you open some tabs).   
2. The buffer window mislead anyone because the file buffer page likes "Tab".

**Comment by kalloc** [comment]
I have the same problem and can reproduce it using fresh `lazyvim` install:
1) my native `nvim` setup, where `tabclose` visually close tabs
https://asciinema.org/a/jDM3xvesCQPwGT5On9n8NX2Dz

2) fresh `lazyvim`, where `tabclose` do something, but visually tabs stay visually unclosed
https://asciinema.org/a/KKDkFl70gNkwYWO6KgtMgMMf5

**Reply by dpetka2001** [reply]
Buffers are shared between tabs. In your personal configuration you must have some plugin that closes the buffers when you close tabs. But that is not default Neovim behavior. Closing a tab/window does not also close the buffer.

**Comment by kalloc** [comment]
@dpetka2001 ok, https://asciinema.org/a/P0Nt5yZNEDF0WKdaaayALI8gp there is empty Neovim without any configs with expected behavior

**Reply by dpetka2001** [reply]
Then it's related to `bufferline.nvim`. If you do `nvim --clean` and then `:tabnew` 4 times and then also close with `:tabclose` you don't see the buffers, because you don't have `bufferline.nvim`. You only see one buffer on each tab. But if you do `:buffers` the buffers are still there shared among all the tabs. It's just that `bufferline.nvim` shows those buffers. If you don't like it, then either read the `bufferline.nvim` docs to see if there's a setting that disables this or disable `bufferline.nvim` as a whole. Either way not a LazyVim problem.

**Comment by kalloc** [comment]
<strike>https://asciinema.org/a/XqcOJdRiiuXaA3O7Ual9d9sxo @dpetka2001 I did and it worked </strike>
ok, will go this way, thanks
